pymc3/model.py

```diff
@@ -920,6 +921,50 @@ def makefn(self, outs, mode=None, *args, **kwargs):
                                    accept_inplace=True,
                                    mode=mode, *args, **kwargs)
 
+    def set_data(self, new_data):
+        """Sets the value of one or more data container variables.
+
+        Parameters
+        ----------
+        new_data : dict
+            New values for the data containers. The keys of the dictionary are
+            the  variables names in the model and the values are the objects
+            with which to update.
+
+        Examples
+        --------
+
+        .. code:: ipython
+
+            >>> import pymc3 as pm
+            >>> with pm.Model() as model:
+            ...    x = pm.Data('x', [1., 2., 3.])
+            ...    y = pm.Data('y', [1., 2., 3.])
+            ...    beta = pm.Normal('beta', 0, 1)
+            ...    obs = pm.Normal('obs', x * beta, 1, observed=y)
+            ...    trace = pm.sample(1000, tune=1000)
+
+        Set the value of `x` to predict on new data.
+
+        .. code:: ipython
+            >>> pm.set_data(new_data={'x': [5,6,9]}, model=model)
```

Comment:
I would change this to also use the `with model` syntax which we're  using almost everywhere else (and remove the note about calling `model.set_data()`).